### PR TITLE
python,python3: cleanup HOST vars and disable pip on host python3

### PR DESCRIPTION
**Pull request by commodo**
Maintainer: me
Compile tested: LEDE trunk
Run tested: N/A

Description:
Cleanup HOST vars.
Disable pip for python3's host build

Signed-off-by: Alexandru Ardelean <ardeleanalex@gmail.com>

### CODE REVIEW

File: lang/python/Makefile [hunk 1]
```diff
@@ -147,13 +147,13 @@ define Build/InstallDev
 		$(PKG_INSTALL_DIR)/usr/include/python$(PYTHON_VERSION) \
 		$(1)/usr/include/
 	$(CP) \
-		$(HOST_BUILD_PREFIX)/lib/python$(PYTHON_VERSION) \
+		$(HOST_PYTHON_LIB_DIR) \
```
Comment (hnyman) [inline lang/python/Makefile]: I applied the 3 patches manually and python2.7 seems to fail at the installdev stage if I get it right:
```
install -d -m0755 /Openwrt/r7800/tmp/stage-python/usr/lib/python2.7/
install -m0644 ./files/python-package.mk ./files/python-host.mk /Openwrt/r7800/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/mk/
cp -fpR /Openwrt/r7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/Python-2.7.12/ipkg-install/usr/include/python2.7 /Openwrt/r7800/tmp/stage-python/usr/include/
cp -fpR /Openwrt/r7800/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/host/lib/python /Openwrt/r7800/build_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/Python-2.7.12/ipkg-install/usr/lib/libpython2.7.so* /Openwrt/r7800/tmp/stage-python/usr/lib/
cp: cannot stat '/Openwrt/r7800/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/host/lib/python': No such file or directory
Makefile:238: recipe for target '/Openwrt/r7800/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/stamp/.python_installed' failed
make[2]: *** [/Openwrt/r7800/staging_dir/target-arm_cortex-a15+neon-vfpv4_musl-1.1.15_eabi/stamp/.python_installed] Error 1
```
To me it looks like the cp command gets the first argument without "2.7" appendix. That should be coming from HOST_PYTHON_LIB_DIR.

**Comment by commodo** [comment]
oh ; right ; PR is open on this branch [Monday coffee still needs to kick in]
testing changes locally now;

**Comment by hnyman** [comment]
hmmm. I tried to compilation with your earlier changes (as of 40 minutes ago). Both python and python3 compiled ok, but python-setuptools gave me error.

But I think that you just refreshed the patches while I was writing this.

(If fixing this gets tricky, it might be easier to split the PR into several pieces and get e.g. the HOST_CFLAGS & double-inclusion guard committed first.)

**Comment by hnyman** [comment]
After your last changes it seems to compile ok (at least in LEDE). So, let's merge this and see the effects...

**Comment by commodo** [comment]
i'm starting to feel like python is becoming like OpenWrt/LEDE core
[in the sense that] i have to start caring more & more about the rest of the python packages

i did not see that coming when i started maintaining python ; but it is an interesting self-discipline lesson

**Comment by hnyman** [comment]
> i'm starting to feel like python is becoming like OpenWrt/LEDE core [in the sense that] i have to start caring more & more about the rest of the python packages

That is to large extent due to Python packages wanting to use Python as the compilation, setup (& installation?) tool. Most other packages use the native Openwrt/LEDE compilation, setup & installation tools, but for some reason the Python ecosystem seems to be built around using Python to compile/setup Python packages. Really snaky.

I wonder if it would be possible to revert to the default package behavior, where packages use the native tools to compile & copy the files after compilation, and dependencies take care of the needed libraries. I am not an active Python user, so I have no idea how realistic that kind of wish is.

Note that the same behavior can be observed to some extent in some other high-level languages like ruby (e.g., discussion about "gem install" in #3454 ). So Python is not alone on that front.

**Comment by commodo** [comment]
it would be interesting to try to reduce dependency on `python` for python packages;

Python-people tend to re-invent the wheel and (re)write their own stuff ; the python hell starts from the distutils (python built-in package), which is needed by python-setuptools, and python-pip which needs python-setuptools
(all 3 meant to do the same thing [make an OS agnostic toolset for installing python packages], but the wheel has been re-invented 3 times and each became popular)

the whole reason of pip (and me maintaining it) is so that people should not have to maintain python packages (which are pure Python source code) ; rather they should be installed/downloaded from pypi [which is an online python packages repo].

(i can't fight [well, it would be unfair of me to fight against] people who want to maintain python packages in OpenWrt ; or at least that was the stance i decided to take on that )

regarding Ruby ; I also see another group of people who like re-inventing the wheel [in another way]

personally, i like Python for doing some quick PoCs/prototypes ; I don't use it much lately ;
I used it  a bit more intensively [for personal stuff] when I became a maintainer ;
then people asked me for help with it ; and i don't mind updating it every once in a while ; tho it's not always smooth

**Comment by hnyman** [comment]
@commodo  @Shulyaka 
After this PR all other python packages seem to compile ok in the LEDE buildbot, but python-dns fails.

When I look at python-dns, I notice that it has different compile section options than the other packages:
python-dns: ``` install --prefix="$(PKG_INSTALL_DIR)/usr" ```
other python packages: ``` install --prefix=/usr --root="$(PKG_INSTALL_DIR)" ```

Could that be the reason?
https://github.com/openwrt/packages/blob/master/lang/python-dns/Makefile

**Comment by commodo** [comment]
it could be ; will try locally ; 
i vaguely remember something about --prefix & --root & something like this

**Comment by Shulyaka** [comment]
@hnyman,
I'm ok with the change, thank you for pointing out.

@commodo,
> the whole reason of pip (and me maintaining it) is so that people should not have to maintain python packages (which are pure Python source code) ; rather they should be installed/downloaded from pypi [which is an online python packages repo].

What if someone maintains a python package only as a dependency for another package? Do you suggest to insert a pip install command to the postinst script? That could be an issue if someone decides to install a locally compiled package offline before setting up upstream internet connection (not a common use case, of course).

**Comment by commodo** [comment]
@Shulyaka that makes sense.
For me, I'm just trying to find the minimal effort path with Python :p

**Comment by commodo** [comment]
this fixes python-dns https://github.com/openwrt/packages/pull/3499

**Pull request merged**